launch/test/launch/substitutions/test_equals_substitution.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the EqualsSubstitution class."""
+
+from launch import LaunchContext
+
+from launch.substitutions import EqualsSubstitution
+from launch.substitutions import PathJoinSubstitution
+
+import os
+
+
+def test_equals_substitution():
+    lc = LaunchContext()
+    assert EqualsSubstitution(None, None).perform(lc) == 'true'
+    assert EqualsSubstitution(None, "something").perform(lc) == 'false'
+    assert EqualsSubstitution(True, True).perform(lc) == 'true'
+    assert EqualsSubstitution(False, False).perform(lc) == 'true'
+    assert EqualsSubstitution(True, False).perform(lc) == 'false'
+    assert EqualsSubstitution(1, True).perform(lc) == 'true'
+    assert EqualsSubstitution(1, 1).perform(lc) == 'true'
+    assert EqualsSubstitution(1, 0).perform(lc) == 'false'
+    assert EqualsSubstitution(1, "1").perform(lc) == 'false'
```

Comment:
I think this could be a source of confusion for users. IIUC, the following substitution will always return `'false'` since substitutions always result in a string:

```py
EqualsSubstitution(LaunchConfiguration('foo'), 1)
```
I think either we should cast non-substitution arguments to strings before doing the comparison _or_ only support substitution types as arguments.

Reply:
Alrighty, I'll need to write some extra logic to ensure `1` and `True` (and `0` and `False`) remain equatable, since I can see that as a case that occurs frequently..

Should a non-zero integer and `True` equate also..? :thinking:
For string comparison reasons I'd like to say no.

Reply:
[See tests](https://github.com/ros2/launch/pull/649/commits/43a9034ee0528219b36dc43ec560b7d85a7f7b85#diff-34ab44869ae2b18792077675afa5660d2bf35fa2531a62efb1a48d5698531a71)

Reply:
> Should a non-zero integer and True equate also..? 

I'd say no as well, since `42 != True` in Python.

Reply:
Ah, gotcha. I got confused because `bool(42) == True`, but I guess that's one conversion too many :grimacing: